### PR TITLE
Master wowl studio vem lpe

### PR DESCRIPTION
**Pull request by kebeclibre**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/notebook/notebook.js [hunk 1]
```diff
@@ -130,6 +140,9 @@ export class Notebook extends Component {
             } else {
                 pages.push([id, v]);
             }
+            if (v.isDisabled) {
+                this.disabledPages.push(k);
```
Comment (aab-odoo) [inline addons/web/static/src/core/notebook/notebook.js:144]: only works when pages given in props, not as slots, right?

File: addons/web/static/src/core/notebook/notebook.xml [hunk 2]
```diff
@@ -5,8 +5,11 @@
...
-                        <a class="nav-link" t-attf-class="{{ navItem[0] === state.currentPage ? 'active' : '' }} {{ props.orientation === 'vertical' ? 'p-3 rounded-0' : '' }} {{ navItem[1].className }}" t-att-name="navItem[1].name" t-on-click.prevent="() => this.activatePage(navItem[0])" href="#" role="tab" tabindex="0" t-esc="navItem[1].title"/>
+                    <li t-foreach="navItems" t-as="navItem" t-key="navItem[0]" class="nav-item flex-nowrap cursor-pointer" t-if="navItem[1].isVisible" t-attf-class="{{ navItem[1].isDisabled ? 'disabled' : '' }}">
+                        <a class="nav-link" t-attf-class="{{ navItem[0] === state.currentPage ? 'active' : '' }} {{ props.orientation === 'vertical' ? 'p-3 rounded-0' : '' }} {{ navItem[1].className }}" t-att-name="navItem[1].name" t-on-click.prevent="() => this.activatePage(navItem[0])" href="#" role="tab" tabindex="0">
+                            <i t-if="props.icons and props.icons[navItem[0]]" t-attf-class="fa {{ props.icons[navItem[0]] }} me-2" />
+                            <t t-esc="navItem[1].title" />
+                        </a>
```
Comment (aab-odoo) [inline addons/web/static/src/core/notebook/notebook.xml:12]: image

File: addons/web/static/src/views/fields/x2many/x2many_field.js [hunk 3]
```diff
@@ -277,11 +277,13 @@ export const x2ManyField = {
             addLabel: attrs["add-label"],
             context: dynamicInfo.context,
             domain: dynamicInfo.domain,
-            viewMode,
             views,
             crudOptions: options,
             string,
         };
+        if (viewMode) {
```
[the remark this answers is not in the document]
Comment (aab-odoo) [reply addons/web/static/src/views/fields/x2many/x2many_field.js:284]: views is optional too I guess, maybe cleaner to remove both

File: addons/web/static/src/core/select_menu/select_menu.js [hunk 4]
```diff
@@ -116,7 +120,7 @@ export class SelectMenu extends Component {
     }
 
     canClear() {
-        return this.props.value != null;
+        return !this.props.required && this.props.value != null;
```
Comment (aab-odoo) [inline addons/web/static/src/core/select_menu/select_menu.js]: ideally this commit should add/edit tests

File: addons/web/static/src/core/select_menu/select_menu.js [hunk 5]
```diff
@@ -89,9 +90,15 @@ export class SelectMenu extends Component {
     get displayValue() {
         if (this.props.value !== undefined) {
             const choices = [...this.props.choices, ...this.props.groups.flatMap((g) => g.choices)];
-            const value = choices.find((c) => c.value === this.props.value);
+            const value = choices.find((c) => {
+                if (typeof c.value === "object" && typeof this.props.value === "object") {
+                    return shallowEqual(c.value, this.props.value);
+                }
+                return c.value === this.props.value;
+            });
```
Comment (aab-odoo) [inline addons/web/static/src/core/select_menu/select_menu.js]: should be tested

File: addons/web/static/src/views/calendar/calendar_controller.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ import { CalendarQuickCreate } from "./quick_create/calendar_quick_create";
 
 import { Component, useState } from "@odoo/owl";
 
-const SCALE_LABELS = {
+export const SCALE_LABELS = {
```
Comment (aab-odoo) [inline addons/web/static/src/views/calendar/calendar_controller.js:18]: weird diff in this commit

File: addons/web/static/src/core/select_menu/select_menu.js [hunk 7]
```diff
@@ -64,6 +66,7 @@ export class SelectMenu extends Component {
         searchable: { type: Boolean, optional: true },
         searchPlaceholder: { type: String, optional: true },
         value: { optional: true },
+        multiSelect: { type: Boolean, optional: true },
```
Comment (aab-odoo) [inline addons/web/static/src/core/select_menu/select_menu.js:69]: this props must be tested

Comment (aab-odoo) [reply addons/web/static/src/core/select_menu/select_menu.js:69]: and documented

File: addons/web/static/src/core/select_menu/select_menu.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /** @odoo-module **/
 
-import { Component, useState, useRef } from "@odoo/owl";
```
Comment (aab-odoo) [inline addons/web/static/src/core/select_menu/select_menu.js:3]: [FIX] **W**eb: -> [FIX] web:

Comment (aab-odoo) [reply addons/web/static/src/core/select_menu/select_menu.js:3]: should this commit be re-targeted to 16.0?

File: addons/web/static/src/core/select_menu/select_menu.js [hunk 9]
```diff
@@ -88,20 +88,25 @@ export class SelectMenu extends Component {
             () => this.onInput(this.inputRef.el ? this.inputRef.el.value.trim() : ""),
             250
         );
+
+        this.selectedChoice = this.getSelectedChoice(this.props);
+        onWillUpdateProps((nextProps) => {
+            if (this.props.value != nextProps.value) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/select_menu/select_menu.js:94]: !==

File: addons/web/static/src/core/select_menu/select_menu.js (hunk 8 above)
Comment (aab-odoo) [reply addons/web/static/src/core/select_menu/select_menu.js:3]: should be tested as well

File: addons/web/static/tests/core/select_menu_tests.js [hunk 10]
```diff
@@ -418,4 +418,158 @@ QUnit.module("Web Components", (hooks) => {
             );
         }
     );
+
+    QUnit.test(
+        "When multiSelect is enable, value is an array of values, mutliple choices should display as selected and tags should be displayed",
```
Comment (aab-odoo) [inline addons/web/static/tests/core/select_menu_tests.js:423]: mutliple

File: addons/web/static/src/views/fields/image/image_field.js [hunk 11]
```diff
@@ -134,6 +134,20 @@ export class ImageField extends Component {
 export const imageField = {
     component: ImageField,
     displayName: _lt("Image"),
+    supportedOptions: [
+        {
+            name: "size",
+            label: _lt("Size"),
+            type: "selection",
+            getChoices() {
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/image/image_field.js:142]: suggestion (not sure about that): like we did for relatedFields, fieldDependencies..., getChoices could simply be `choices` which can be an array (like for this usecase) or a function returning an array for dynamic cases like the one below

File: addons/web/static/src/core/notebook/notebook.xml (hunk 2 above)
Comment (Iucapad) [reply addons/web/static/src/core/notebook/notebook.xml:12]: done

File: addons/web/static/src/views/calendar/calendar_controller.js (hunk 6 above)
Comment (Iucapad) [reply addons/web/static/src/views/calendar/calendar_controller.js:18]: it is used by the calendar editor, but has nothing to do with the changes of this commit -> moved in another one

File: addons/web/static/src/core/select_menu/select_menu.js (hunk 7 above)
Comment (Iucapad) [reply addons/web/static/src/core/select_menu/select_menu.js:69]: I'll write a PR of documentation to update SelectMenu and Notebook with the newest props available

**Comment by aab-odoo** [comment]
robodoo rebase-ff r+

**Comment by Xavier-Do** [comment]
@robodoo override=ci/style

**Pull request closed without merging**